docs/Developers/Git_Basics.rst

```diff
@@ -70,7 +70,10 @@ Creating a new branch
 ---------------------
 
 To start working on an issue, you first need to create a new branch where you
-will work.
+will work. You should always have a clean and working branch which
+is considered to be the
+"master" branch. Also, developing on the master branch
+can potentially cause issues.
```

Comment:
Maybe you can add ` The policy for Master is that it is the stable release. Checking in small incremental changes into this means you don't have something stable at all times` which states what issues development on the master branch can cause.